api_app/analyzers_manager/observable_analyzers/dns0_names.py

```diff
@@ -0,0 +1,212 @@
+
+        if not hasattr(self, "_api_key") or not self._api_key:
+            raise AnalyzerRunException("No API key specified")
+
+        self._validate_params()
+
+    def run(self):
+        params = self._create_params()
+        headers = {
+            "Authorization": f"Bearer {self._api_key}",
+            "Accept": "application/json",
+        }
+
+        response = requests.get(self.base_url, params=params, headers=headers)
+        try:
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            raise AnalyzerRunException(e)
+
+        return response.json()
+
+    def _validate_params(self):
+        if hasattr(self, "fuzzy") and any(
+            fuzzy_params not in _supported_fuzzy_params for fuzzy_params in self.fuzzy
+        ):
+            raise AnalyzerConfigurationException(
+                "Fuzzy type not supported! "
+                "The list of supported fuzzy is at: "
+                "https://docs.dns0.eu/dns-api/names#fuzziness"
+            )
+
+        if hasattr(self, "sort") and self.sort not in _supported_sort_types:
+            raise AnalyzerConfigurationException(
+                f"Sort type {self.sort} not supported! "
+                f"Available sort types are: {_supported_sort_types}"
+            )
+
+        if hasattr(self, "format") and self.format not in _supported_format_types:
+            raise AnalyzerConfigurationException(
+                f"Format type {self.format} not supported! "
+                f"Available format types are: {_supported_format_types}"
+            )
+
+        if (
+            hasattr(self, "limit")
+            and not _min_limit_value < self.limit <= _max_limit_value
+        ):
+            raise AnalyzerConfigurationException(
+                f"{self.limit} is out of bound! "
+                f"Max value is {_max_limit_value}, min value is {_min_limit_value}"
+            )
+
+        if hasattr(self, "offset") and self.offset < _min_offset_value:
+            raise AnalyzerConfigurationException(
+                f"{self.offset} can't be below {_min_offset_value}"
+            )
+
+    @staticmethod
+    def convert_date_type(date_string):
+        if not date_string:
+            return False
+
+        return (
+            DNS0Names.convert_unix_timestamp(date_string)
+            or DNS0Names.convert_relative_date(date_string)
+            or DNS0Names.convert_date(date_string)
+        )
+
+    @staticmethod
+    def convert_relative_date(date):
+        # accepts string matching the format:
+        # - at the beginning
+        # a number
+        # a character indicating Year, Month or Day
+        pattern = re.compile(r"-\d+[YMD]")
+        if match := pattern.match(date):
+            return match.group()
+        return False
+
+    @staticmethod
+    def convert_date(date):
+        try:
+            parsed = dateutil_parser.parse(date).strftime("%Y-%m-%d")
+            return parsed
+        except Exception:
```

Comment:
you should catch only the exception that you need to catch, not a generic one because the risk is that we could miss other unexpected errors in this way

Reply:
i think that the whole logic of date conversion can be improved. Any suggestion on how to refactor that ?